Enable synthetically delaying FCP

sndrs opened the pull request:
_n.b. relies on #7639..._

## What does this change?

Enables synthetically delaying rendering of a page by adding a `#bork-fcp-xxx` hash flag, where `xxx` is the time in ms by which to delay showing the body.

## Why?

This will allow us to test various FCP slowdowns

mxdvl reviewed (approved):
…

<br/>

<br/>

<br/>

<br/>

<br/>

<br/>

This looks great!

mxdvl commented on dotcom-rendering/src/web/bork/fcp.ts:
```diff
@@ -0,0 +1,19 @@
+			const root = document.documentElement;
+			root.style.setProperty(
+				'--bork-fcp-amount',
+				hash.replace(hashKey, '') + 'ms',
```
Should this be an additional delay once loaded, or an absolute time from the beginning of page load?

jamesgorrie reviewed (commented):
Super smart, just wondering if we should be consistent in the way we enable these tests.

sndrs commented on dotcom-rendering/src/web/server/pageTemplate.ts:
```diff
@@ -319,6 +324,21 @@ https://workforus.theguardian.com/careers/product-engineering/
 				${css}
 				<link rel="stylesheet" media="print" href="${ASSET_ORIGIN}static/frontend/css/print.css">
+  				<style>
+					@keyframes show-me {
```
yeah very good point, thanks

mxdvl commented on dotcom-rendering/src/web/server/pageTemplate.ts:
```diff
@@ -300,10 +301,10 @@ https://workforus.theguardian.com/careers/product-engineering/
 				</script>
 
 				${bork ? `<script>(${fid.toString()})()</script>` : ''}
+				${bork ? `<script>(${fcp.toString()})();</script>` : ''}
```
if you like it when things line up…

```suggestion
				${bork ? `<script>(${fcp.toString()})()</script>` : ''}
```

mxdvl commented on dotcom-rendering/src/web/bork/fcp.ts:
```diff
@@ -0,0 +1,19 @@
+export const fcp = (): void => {
```
It might be good to have a little description, possibly [pointing to the CLS definition](https://web.dev/cls/).

sndrs commented:
> Super smart, just wondering if we should be consistent in the way we enable these tests.

💯 

i opened this before we added the switch, i've updated the code to use the same switch as #7639
